server/src/main/java/org/folio/reservoir/module/impl/ModuleCacheImpl.java

```diff
@@ -24,33 +24,45 @@ private ModuleCacheImpl() { }
 
   private class CacheEntry {
     private final Module module;
-    private final JsonObject config;
+    private final CodeModuleEntity entity;
 
-    CacheEntry(Module module, JsonObject config) {
+    CacheEntry(Module module, CodeModuleEntity entity) {
       this.module = module;
-      this.config = config;
+      this.entity = entity;
     }
 
   }
 
+  private Module createInstance(String type) {
+    if (type == null) {
+      type = "";
+    }
```

Comment:
```suggestion
    if (type == null) {
      type = "javascript";
    }
```

Let's leave special values, like "" out of this. JSON can represent that fine with an omitted property.